packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -657,6 +658,8 @@ export class BackgroundScript {
     }
 
     const receiver = stream.getPaymentPointer()
+    const {assetCode, assetScale, exchangeRate} = stream.getAssetDetails()
```

Comment:
Need to handle the rare case that assetCode/assetScale etc haven't been sent yet no?

Reply:
True, I have been wondering how. Maybe disable tipping if the stream connection cannot be established?

Reply:
That could work. You could do that on the popup side using existing state, hiding the tipping button if the balance is non-zero.

Reply:
s/if/unless